### DISCUSSION TITLE
Ifdef loader for parcel v2?

### DISCUSSION THREAD

**Discussion by Cristy94** [parcel-bundler/parcel#5651] [Q&A]
I need to conditionally builds having different features sets (eg. TRIAL, REGULAR, PREMIUM), so I have to exclude parts of the code for specific builds.

With webpack I was doing this using [ifdef-loader](https://github.com/nippur72/ifdef-loader), is there such a loader for Parcel v2?
If not, what is the best way to conditionally exclude certain parts of the code from being included in the build?

For example:

```
function reg() {

}

// #if-def PREMIUM-VERSION
function premium {
    // This will only be included in the build if PREMIUM-VERSION environment variable is set
}
// #endif
```

**Comment by mischnic** [comment]
If you do something like `if(process.env.FEATURES === "premium")` and set that env variable accordingly, Terser will completely remove the unused if statements.

**Reply by Cristy94** [reply]
I tried it, but there were still references to the imported file in the bundle as the import is always at the top (can not be wrapped in if statements).

**Reply by Cristy94** [reply]
Oh, and also if I use `.env.premium` and `.env.prod`, how can I tell parcel to build using the different env files? Is it only possible by setting `NODE_ENV` beforehand? eg. `NODE_ENV=premium` ?

**Reply by Cristy94** [reply]
@mischnic Any news on this? How can I tell parcel to use a specific .env file when building, I didn't find anything in the docs.

Also, are you sure the if statements and all included references are removed? I tried it and the Premium.ts class was still loaded.

**Reply by mischnic** [reply]
1. https://github.com/parcel-bundler/parcel/issues/3435#issuecomment-787110173
2. Can you provide an example

**Reply by Cristy94** [reply]
1. Thanks, so I can do `NODE_ENV='prod.premium'`
2. The project is really complex, but I only have a React component being conditionally included:

```
// Comp1.tsx
export Comp1.ts = () => {}

// Main.tsx
if(process.env.premium) {
    <Comp1 />
}
```

If the env condition is false, the application works as expected but I can still see references to `Comp1` and its source in the final bundle.

**Reply by Cristy94** [reply]
Ok, so I managed to make it work but there is one error in production:

To load the custom .env file I set this `process.env.NODE_ENV = 'prod.premium'`.

The build works, but because `NODE_ENV` is not `production` I get an error:

```
You are currently using minified code outside of NODE_ENV === "production". This means that you are running a slower development build of Redux. You can use loose-envify (https://github.com/zertosh/loose-envify) for browserify or setting mode to production in webpack (https://webpack.js.org/concepts/mode/) to ensure you have the correct code for your production build.
```

It would be nice if we could pass .env files like `PARCEL_ENV = .env.prod.premium`

**Reply by Cristy94** [reply]
Actually I fixed the production issue by adding this in my .env file:

`NODE_ENV=production`

So I set `NODE_ENV` before build to select the .env file, then inside the .env file I overwrite the `NODE_ENV` to production.
